pymatgen/io/aims/inputs.py

```diff
@@ -0,0 +1,609 @@
+        edges (Sequence[Sequence[float]]): Specifies the edges of a cube: dx, dy, dz
+            dx (float): The length of the step in the x direction
+            dy (float): The length of the step in the y direction
+            dx (float): The length of the step in the x direction
+        points (Sequence[int] or tuple[int, int, int]): The number of points
+            along each edge
+        spinstate (int): The spin-channel to use either 1 or 2
+        kpoint (int): The k-point to use (the index of the list printed from
+            `output k_point_list`)
+        filename (str): The filename to use
+        format (str): The format to output the cube file in: cube, gOpenMol, or xsf
+        elf_type (int): The type of electron localization function to use (
+            see FHI-aims manual)
+    """
+
+    type: str = field(default_factory=str)
+    origin: Sequence[float] | tuple[float, float, float] = field(default_factory=lambda: [0.0, 0.0, 0.0])
+    edges: Sequence[Sequence[float]] = field(default_factory=lambda: 0.1 * np.eye(3))
+    points: Sequence[int] | tuple[int, int, int] = field(default_factory=lambda: [0, 0, 0])
+    format: str = "cube"
+    spinstate: int | None = None
+    kpoint: int | None = None
+    filename: str | None = None
+    elf_type: int | None = None
+
+    def __post_init__(self) -> None:
+        """Check the inputted variables to make sure they are correct
+
+        Raises:
+            ValueError: If any of the inputs is invalid
+        """
+        split_type = self.type.split()
+        if split_type[0] in ALLOWED_AIMS_CUBE_TYPES:
+            if len(split_type) > 1:
+                msg = f"Cube of type {split_type[0]} can not have a state associated with it"
+                raise ValueError(msg)
+        elif split_type[0] in ALLOWED_AIMS_CUBE_TYPES_STATE:
+            if len(split_type) != 2:
+                msg = f"Cube of type {split_type[0]} must have a state associated with it"
+                raise ValueError(msg)
+        else:
+            raise ValueError("Cube type undefined")
+
+        if self.format not in ALLOWED_AIMS_CUBE_FORMATS:
+            raise ValueError(f"{self.format} is invalid. Cube files must have a format of {ALLOWED_AIMS_CUBE_FORMATS}")
+
+        if self.spinstate is not None and (self.spinstate not in [1, 2]):
+            raise ValueError("Spin state must be 1 or 2")
+
+        if len(self.origin) != 3:
+            raise ValueError("The cube origin must have 3 components")
+
+        if len(self.points) != 3:
+            raise ValueError("The number of points per edge must have 3 components")
+
+        if len(self.edges) != 3:
+            raise ValueError("Only three cube edges can be passed")
+
+        for edge in self.edges:
+            if len(edge) != 3:
+                raise ValueError("Each cube edge must have 3 components")
+
+        if self.elf_type is not None and self.type != "elf":
```

Comment:
```suggestion
        if self.elf_typ not in (None, "elf"):
```

Reply:
Fixed, inline since I used `elf_type` and not `elf_typ` to match the FHI-aims documentation.

Reply:
This was actually correct as it is checking `elf_type` and `type`.